src/main/java/htsjdk/samtools/util/Histogram.java

```diff
@@ -317,14 +320,24 @@ public double getPercentile(final double percentile) {
         if (percentile <= 0) throw new IllegalArgumentException("Cannot query percentiles of 0 or below");
         if (percentile >= 1) throw new IllegalArgumentException("Cannot query percentiles of 1 or above");
 
-        double total = getCount();
+        values().stream()
+                .filter(b -> b.getValue() < 0)
+                .findFirst()
+                .ifPresent(b -> {
+            throw new IllegalArgumentException("Cannot calculate Percentile when negative counts are present " +
```

Comment:
I think these should be `IllegalStateException`s because they depend on the state of the `Histogram` object and not the parameters the function is being called with.

Reply:
reasonable.